apps/gp2-frontend/src/working-groups/__tests__/EditResource.test.tsx

```diff
@@ -0,0 +1,204 @@
+      id: workingGroup.id,
+      userId: '23',
+      role: 'Administrator',
+      route: gp2Routing
+        .workingGroups({})
+        .workingGroup({ workingGroupId: workingGroup.id })
+        .resources({}).$,
+      children: <EditResource {...props} />,
+    });
+
+    expect(screen.getByRole('button', { name: /Save/i })).toBeInTheDocument();
+  });
+
+  it('see if content is empty', async () => {
+    const workingGroup = gp2Fixtures.createWorkingGroupResponse();
+    workingGroup.members = [
+      {
+        userId: '23',
+        firstName: 'Tony',
+        lastName: 'Stark',
+        role: 'Lead',
+      },
+    ];
+    delete workingGroup.resources;
+    mockGetWorkingGroup.mockResolvedValueOnce(workingGroup);
+
+    const resources = workingGroups({})
+      .workingGroup({ workingGroupId: workingGroup.id })
+      .resources({}).$;
+
+    const props = {
+      workingGroupId: workingGroup.id,
+      workingGroup,
+      backHref: resources,
+      updateWorkingGroupResources: jest.fn(),
+    };
+
+    await renderWorkingGroupDetail({
+      id: workingGroup.id,
+      userId: '23',
+      role: 'Administrator',
+      route: gp2Routing
+        .workingGroups({})
+        .workingGroup({ workingGroupId: workingGroup.id })
+        .resources({}).$,
+      children: <EditResource {...props} />,
+    });
+
+    expect(
+      screen.getByRole('textbox', { name: 'Resource Type (required)' }),
+    ).toHaveValue('');
+  });
+
+  it('can submit a form when form data is valid for editing', async () => {
+    const workingGroup = gp2Fixtures.createWorkingGroupResponse();
+    workingGroup.members = [
+      {
+        userId: '23',
+        firstName: 'Tony',
+        lastName: 'Stark',
+        role: 'Lead',
+      },
+    ];
+
+    const title = 'example42 title';
+    const type = 'Note';
+
+    const resources = workingGroups({})
+      .workingGroup({ workingGroupId: workingGroup.id })
+      .resources({}).$;
+
+    const props = {
+      workingGroupId: workingGroup.id,
+      workingGroup,
+      backHref: resources,
+      updateWorkingGroupResources: jest.fn(),
+    };
+
+    mockGetWorkingGroup.mockResolvedValueOnce(workingGroup);
+    mockPutWorkingGroupResources.mockResolvedValueOnce(workingGroup);
+    await renderWorkingGroupDetail({
+      id: workingGroup.id,
+      userId: '23',
+      role: 'Administrator',
+      route: gp2Routing
+        .workingGroups({})
+        .workingGroup({ workingGroupId: workingGroup.id })
+        .resources({}).$,
+      children: <EditResource {...props} />,
+    });
+
+    const typeBox = await screen.findByRole('textbox', { name: /type/i });
+    userEvent.type(typeBox, `${type}{enter}`);
+    const titleBox = screen.getByRole('textbox', { name: /title/i });
+    userEvent.type(titleBox, title);
+    const saveButton = screen.getByRole('button', { name: /save/i });
+    userEvent.click(saveButton);
+
+    await waitFor(() => expect(saveButton).toBeEnabled());
+  });
+});
```

Comment:
We are missing a test for multiple resources

Reply:
and we need to consider coverage over submitting all fields?

externalLink and description.

Reply:
Done